### ISSUE TITLE
WARNING: This way of doing signal handling is now deprecated

### ISSUE THREAD

**Issue by DannyBen** [resque/resque#1366]
Not sure what I am doing wrong, or if it is an actual issue.

#### TL;DR:

When running `QUEUE=* rake environment resque:work` I am getting this warning:

**WARNING: This way of doing signal handling is now deprecated. Please see http://hone.heroku.com/resque/2012/08/21/resque-signals.html for more info.**

The only other issue referencing it I found is old (but open) - ticket #999 

#### Some background:

In an older rails app I have used `gem "resque", "~> 2.0.0.pre.1", github: "resque/resque"` in my gemfile.
As I remember, installation and running it was easy and flawless.
Also - in this older version, running resque was done with simply running `resque work`

This was nice and easy and I wonder why it was removed in the new version (or if I forgot to configure something...).

With a newer rails app I am doing, I simply followed the instructions in the README and used `gem 'resque'` and added the rake task etc. Then, when running the rake task I get this warning.

#### Additional notes:

As a side note, I think the wiki should include short, clean and concise installation and 'getting started' instructions per platform. The README contains all sorts of topics in a way that probably makes sense to some people, but not to those who come to "get started" with it.

Also - and I am saying it just because I know it *can* be easier - I think the whole running it with a rake task, preceded by a load of environment variables does not do it justice and for me is a bad turn in resque's journey (assuming it is indeed the intended design, and not a mistake on my part).

The resque command line that handled everything including running workers was a far superior design choice in my mind. At least from a user standpoint.

**Comment by steveklabnik** [comment]
Hey @DannyBen 

sorry for not getting back to you earlier. I wasn't sure myself.

I think the issue here is that in your older app you were using the 2.0 version, but in the newer one, you're using 1.x. Can you give me any more ideas about that? I thought that we had dealt with the signal handling thing, but I'm not sure.

**Comment by DannyBen** [comment]
That's perfectly fine, I know the real life sometimes takes you away from the open source projects.

Well, now trying to refresh my own memory, I am getting confused myself.
In the old app, I used version 2, by bringing it directly from github

In the new app, I figured I will try to follow the README instructions to the letter, so I just used `gem 'resque'`.

But now, when I go to the [resque page on rubygems](https://rubygems.org/gems/resque/versions/1.25.2) I actually see it is severely outdated.

So - maybe lets resolve the simpler confusions first:

1. Is there a reason the gem is not updated on rubygems?
2. Shouldn't the README point that out, and instruct readers to use github as a source when including the gem?
3. What is the actual line we need to use in the Gemfile to get the latest stable release? Is it still `gem "resque", "~> 2.0.0.pre.1", github: "resque/resque"`

**Comment by steveklabnik** [comment]
> Is there a reason the gem is not updated on rubygems?

There has not been a 1-x release in a long time. Work went into 2.0 instead, which is now thrown away. There will be another 1-x release at some point. 2.0 is never happening.

Mature, stable software doens't need a new release every few weeks to be working well.

> What is the actual line we need to use in the Gemfile to get the latest stable release?

`gem "resque"`.

**Comment by DannyBen** [comment]
Oh.... thats a shame... So I am using version 2.0 which is actually not the recommended version....

So, then we are back to square one.
Using "gem 'resque'" generates the above mentioned error for me.

**Comment by steveklabnik** [comment]
Can you give me any other sort of context for this? Does it happen with an empty app? What code ends up generating the warning?

**Comment by DannyBen** [comment]
I will try.

**Comment by DannyBen** [comment]
Ok. Here are the findings.

Yes, it seems to also happen with a squeaky clean rails app.

**Some context:**

```
$ lsb_release -d
Description:    Ubuntu 14.04.3 LTS

$ ruby -v
ruby 2.2.1p85 (2015-02-26 revision 49769) [x86_64-linux]

$ rails -v
Rails 4.2.5

$ rvm -v
rvm 1.26.11 (latest)

$ bundle show resque
/home/vagrant/.rvm/gems/ruby-2.2.1/gems/resque-1.25.2

$ QUEUE=default rake environment resque:work
WARNING: This way of doing signal handling is now deprecated. Please see http://hone.heroku.com/resque/2012/08/21/resque-signals.html for more info.
```

This happens on my actual app, and below are the steps I took to also try it with a fresh app:

1. `rails new resque-bug`
2. Add `gem 'resque'` to Gemfile, and run `bundle`
3. Add 'require 'resque/tasks'` to `lib/tasks/resque.rake`
4. Run `QUEUE=default rake environment resque:work1`

**Comment by steveklabnik** [comment]
Awesome, thanks so much. 

I am not sure when I will have time to investigate this exactly, but will try to soon. Warnings in such a default thing is.. suboptimal.

**Comment by Jacobkg** [comment]
@steveklabnik Reading the blog post in the warning (http://hone.herokuapp.com/resque/2012/08/21/resque-signals.html) it appears that this was part of a plan to move to a different signal trapping paradigm in the next major version. The warning appears for everyone who doesn't set `TERM_CHILD=1` when they start their workers.

**Comment by Jacobkg** [comment]
@steveklabnik If there is no longer a clear plan to see this (breaking) change through, my recommendation would be to remove the deprecation warning for 1.26 and just continue to support the default and `TERM_CHILD=1` options indefinitely. We may want to add something to the README about the new option as well.

**Comment by Jacobkg** [comment]
@steveklabnik Do we want to keep the deprecation warning for 1.26 or remove it and continue to support both the default and TERM_CHILD=1 options indefinitely?

**Comment by steveklabnik** [comment]
I think removing the deprecation warning makes sense until we're actually sure what we want to do here. If there's no actual plan to deprecate, no reason for the warning.

**Comment by chrisccerami** [comment]
With the deprecation removed, is this issue good to be closed?

**Comment by DannyBen** [comment]
If the question is directed at `world`, then I say yes :)

**Comment by steveklabnik** [comment]
Yes, it is. Thanks!
